src/App.tsx

```diff
@@ -559,24 +605,120 @@ function App() {
+    setExportQuestion(question);
+  }, [exportView, loadedQuestionCorrect, question]);
+
+  const pngExportView = useCallback((name: string) => {
+    if (!database || exportQuery) {
+      return;
+    }
+    const view = views.find(v => v.name === name);
+    if (!view) {
+      return;
+    }
+    setExportView(view);
+  }, [database, exportQuery, views]);
+
+  useEffect(() => {
+    if (!exportRendererRef.current || exportingStatus >= 1) {
+      return;
+    }
+
+    // TODO: I'm not really a fan of this solution but without it the browser crashes due to downloading an extreme amount of files
+    setExportingStatus(1);
+
+    // View
+    if (exportView) {
+      toPng(exportRendererRef.current, { 
+        canvasWidth: exportRendererRef.current.clientWidth,
+        width: exportRendererRef.current.clientWidth,
+        canvasHeight: exportRendererRef.current.clientHeight,
+        height: exportRendererRef.current.clientHeight,
+        pixelRatio: 1 
+      }).then((dataUrl) => {
+        const link = document.createElement('a');
+        link.download = `validator_${exportView.name}.png`;
+        link.href = dataUrl;
+        link.click();
+        setExportView(undefined);
+        setExportingStatus(0);
+      });
+      return;
+    }
+
+    // Question
+    if (!question || !exportQuery || !exportQuestion) {
+      return;
+    }
+
+    const exportRenderer = exportRendererRef.current;
+    toPng(exportRenderer, { 
+      canvasWidth: exportRenderer.clientWidth,
+      width: exportRenderer.clientWidth,
+      canvasHeight: exportRenderer.clientHeight,
+      height: exportRenderer.clientHeight,
+      pixelRatio: 1 
+    }).then((dataUrl) => {
+        const link = document.createElement('a');
+        link.download = `validator_${question.id}_${question.category.display_number}${question.display_sequence}.png`;
+        link.href = dataUrl;
+        link.click();
+        setExportQuestion(undefined);
+        setExportQuery(undefined);
+        setExportingStatus(0)
+      }
+    );
+  }, [evaluatedQuery, exportQuery, exportRendererRef, getTheme, isDarkMode, exportingStatus, question, resetResult, setTheme, exportQuestion, exportView]);
+
   return (
     <div className="App">
+      {exportQuestion && exportQuery && <ExportRenderer query={{isCorrect: isCorrectResult(exportQuestion.evaluable_result, evalSql(exportQuery)), question: exportQuestion, code: exportQuery, result: evalSql(exportQuery)}} ref={exportRendererRef} />}
+      {exportView && <ExportRenderer view={{view: exportView, result: evalSql(`SELECT * FROM ${exportView.name}`)}} ref={exportRendererRef} />}
       <header className="App-header">
         <div className='my-2'></div>
         <ThemeToggle setTheme={setTheme} isDarkMode={isDarkMode}></ThemeToggle>
         <h1 className='text-6xl font-semibold my-3'>DB SQL Validator</h1>
-        <img src={isDarkMode() ? db_scheme_dark : db_scheme_light} className="DB-Layout w-full max-w-xl" alt="Database Layout" />
```

Comment:
wrong PR to fix this but accidently set width & height due to two conflicting commits